### DISCUSSION TITLE
msvcrtd.lib(exe_main.obj) : error LNK2019: unresolved external symbol main referenced in function

### DISCUSSION THREAD

**Discussion by ghost** [nokeedev/gradle-native#613] [Q&A]
I added this line to build.gradle file

       linkTask.linkerArgs.add("msvcrt${linkTask.name.toLowerCase().contains('debug') ? 'd' : ''}.lib")

Error :

    > Task :psfe:linkDebugWindows FAILED
    msvcrtd.lib(exe_main.obj) : error LNK2019: unresolved external symbol main referenced in function "int __cdecl invoke_main(void)" (?invoke_main@@YAHXZ)
    Caching disabled for task ':psfe:linkDebugWindows' because:
    D:\a\piv\piv\psfe\build\lib\main\debug\windows\psfe.dll : fatal error LNK1120: 1 unresolved externals
    Build cache is disabled

![mscvrt](https://user-images.githubusercontent.com/104821728/167054756-9f390213-5bb2-4436-8a77-5d20ed69596e.PNG)

How can I solve this error ?

**Comment by lacasseio** [comment]
Please use the discussion for questions. Issues are for feature requests, bugs, or other things that I can act upon to improve the Nokee plugins.

**Comment by lacasseio** [comment]
Try in the future to include more information. There isn't much I can help with this error as it's simply a plain linker error which heavily depends on the build configuration as well as the source code. It would be much better if the smallest complete project which reproduces the problem was included. I can't do that step for you as it will most certainly not represent your project. I can also add that most compiler/linker issue has already a lot of information online. A quick search points to [this thread from Microsoft social](https://social.msdn.microsoft.com/Forums/vstudio/en-US/2dc31c7c-5e92-4afc-9d13-bc3a8bfbe2b6/msvcrtdlibexemainobj-error-lnk2019-unresolved-external-symbol-main-referenced-in-function?forum=vcgeneral). Despite showing the same error as you mention here, there is no way to know if the solution is correct due to the lack of context.

I suggest [reading our small support guideline to help us help you](https://github.com/nokeedev/.github/blob/main/SUPPORT.md).

**Reply by ghost** [reply]
Error : https://github.com/Pradeepjanapriya/withframework/runs/6339009830?check_suite_focus=true#step:9:246

build.gradle file : https://github.com/Pradeepjanapriya/withframework/blob/main/hello/build.gradle#L62

How can I solve this error ?

**Reply by lacasseio** [reply]
Pedro has some good points 👇 . You could try without line https://github.com/Pradeepjanapriya/withframework/blob/main/hello/build.gradle#L64 and https://github.com/Pradeepjanapriya/withframework/blob/main/hello/build.gradle#L62. It seems that you are instructing the toolchain to use a Windows-specific entrypoint which is not provided. In theory, the default entry point should be good enough.

**Comment by pedrolamarao** [comment]
It should not be necessary to explicitly link with Microsoft's C runtime. The toolchain is expected to link with the appropriate runtime variant based on options such as `/MD`.

**Reply by pedrolamarao** [reply]
It should also not be necessary to explicitly set the subsystem when linking a DLL. The default subsystem should be correct.
